src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -184,22 +184,438 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+        # --- basic consistency for per-nucleus arrays ---
+        if self.z_core is not None and self.max_ang_mom_plus_1 is not None:
+            if len(self.z_core) != len(self.max_ang_mom_plus_1):
+                logger.error(
+                    'ECP per-nucleus arrays must have equal length: len(z_core) != len(max_ang_mom_plus_1).'
+                )
+
+        # --- ecp_num-driven arrays must match lengths ---
+        if self.ecp_num is not None:
+            # List of (attribute, attribute_name) tuples for checking
+            projector_arrays = [
+                (self.nucleus_index, 'nucleus_index'),
+                (self.ang_mom, 'ang_mom'),
+                (self.exponent, 'exponent'),
+                (self.coefficient, 'coefficient'),
+                (self.power, 'power'),
+            ]
+
+            for arr, name in projector_arrays:
+                if arr is None:
+                    logger.error('ECP array name is missing while ecp_num is set.')
+                else:
+                    if len(arr) != self.ecp_num:
+                        logger.error('Length of ECP array name must equal ecp_num.')
+
+    @set_not_normalized
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+
+class AtomicOrbitals(ArchiveSection):
+    """
+    Expanded **atomic orbital (AO) layer** associated with a specific
+    `AtomCenteredBasisSet`.
+
+    While `AtomCenteredFunction` defines the *contracted shells* (basis
+    functions before expansion), this section describes the actual AO
+    functions that the code constructs from them. It captures the
+    one-to-one mapping between the mathematical basis definition and the
+    internal AO list used in SCF/post-SCF methods.
+
+    Typical use cases:
+      - Recording whether the code expanded shells in **Cartesian** or
+        **spherical harmonic** form.
+      - Tracking the **per-AO normalization factors** (to reconcile
+        conventions across codes).
+      - Providing an explicit **AO→shell mapping** for downstream
+        population analysis, MO coefficients, or density matrices.
+    """
+
+    num = Quantity(type=np.int32, description='Total number of AOs (contracted).')
+    cartesian = Quantity(type=bool, description='True=Cartesian, False=Spherical.')
+    shell_index = Quantity(
+        type=np.int32,
+        shape=['num'],
+        description="For each AO: index of AtomCenteredFunction (the 'shell') it belongs to.",
+    )
+    normalization = Quantity(
+        type=np.float64,
+        shape=['num'],
+        description='Per-AO factor to reconcile code conventions.',
+    )
 
 
 class AtomCenteredBasisSet(BasisSetComponent):
     """
-    Defines an atom-centered basis set.
+    Defines an **atom-centered basis set** for quantum chemistry calculations.
+    Unlike plane-wave methods, these expansions are typically built around each atom's
```

Comment:
```suggestion
    These expansions are typically built around each atom's
```

I wouldn't try to draw comparisons or distinctions between different techniques for mostly different communities. Each can stand on its own.
Btw, in practice PW methods have a dedicated approach for describing the core electrons, be it pseudopotentials or LAPW. PW is just used for describing the valence electrons at a system-level.